### ISSUE TITLE
Revert Transform Pipeline work back to passes

### ISSUE THREAD

**Issue by rengolin** [plaidml/tpp-mlir#226]
After long discussions with the whole team, the consensus is that there isn't a clear path through transforms that allows us to work together as a team. 

Working on transforms have a number of advantages: higher precision matches; fine grained control of transforms in specific ops; less need for cleanups, But it has a larger number of drawbacks: sequence is hand-coded for each test; no simple path to auto-generation; invalidation of IR (bufferization, canonicalization, lowering) break composition; matches aren't precise enough (we still need to add tags).

But the most important drawback is teamwork. Small changes in the transform semantics have knock-on effect on tests and implementation and it's hard to progress without creating more and more transforms. This has two main problems: first, it's hard to share knowledge of the current state across the team because the state changes too rapidly; and second, it leads to too many rebases on other developers' code and has taken weeks of work in our project for no good reason.

In the end, we're still not clear on how we're going to do any of the critical feature we need, such as automatic detection of tiles and block sizes, multiple matches interoperating with propagation of previous transforms in a larger model. With our primary goal being end-to-end model execution of Python models, having to manually add transform blocks is a non-starter.

So, while we'll still retain a research side to see how it could all work on a separate branch/fork, the `main` branch needs to be entirely pass based.

This issue is about bringing back the deleted passes and moving all tests to pass-based before we can continue working in parallel without having to rebase our local work every week.

**Comment by chelini** [comment]
I don't think it is fair to assign all this work to me.

**Comment by adam-smnk** [comment]
We could split it into some smaller objectives and work on a separate feature branch if we want a single PR to revert everything.

**Comment by rengolin** [comment]
Working on a separate branch is a good idea, but we need to make sure it doesn't live for longer than a week or so, or the rebase on the branch will become harder and harder.

**Comment by KavithaTipturMadhu** [comment]
> We could split it into some smaller objectives and work on a separate feature branch if we want a single PR to revert everything.

Plus one on this, I can pitch in if we break the problem into smaller components we're moving to pass infrastructure.

**Comment by rengolin** [comment]
Awesome! I created a new branch [pass-pipeline](https://github.com/plaidml/tpp-mlir/tree/pass-pipeline) that we can submit PRs to, starting with the [existing one](https://github.com/plaidml/tpp-mlir/pull/225).

This means all my comments can be turned to a checklist in this issue, and we tackle them in parallel, until completion.

**Comment by rengolin** [comment]
I volunteer to do the rebase when new PRs are merged to `main`.

**Comment by rengolin** [comment]
@chelini After #225, what else do we need to do to close this issue?

**Comment by chelini** [comment]
I added back all the passes we used to have. I will double check on Monday, but should be it. It is a good starting point to build a more complete pass pipeline.

**Comment by rengolin** [comment]
The PR was merged, and I created a new milestone for the wider goal of moving everything back to passes (which this is now part of), so I'll close this and we can open new issues on the following items and assign the milestone to it.
